chain/network/src/config.rs

```diff
@@ -288,22 +288,54 @@ impl NetworkConfig {
                 }
             }
         }
+
+        // Create validator config with proper proxies
+        let validator = ValidatorConfig {
+            signer: validator_signer,
+            proxies: if !cfg.public_addrs.is_empty() {
+                ValidatorProxies::Static(cfg.public_addrs.clone())
+            } else {
+                ValidatorProxies::Dynamic(cfg.trusted_stun_servers.clone())
+            },
+        };
+
+        let node_addr = match cfg.addr.as_str() {
+            "" => None,
+            addr => {
+                Some(tcp::ListenerAddr::new(addr.parse().context("Failed to parse SocketAddr")?))
+            }
+        };
+
+        // Parse blacklist from config
+        let blacklist = cfg
+            .blacklist
+            .iter()
+            .map(|e| e.parse::<blacklist::Entry>())
+            .collect::<Result<Vec<_>, _>>()
+            .context("failed to parse blacklist")?
+            .into_iter()
+            .collect::<blacklist::Blacklist>();
+
+        // Determine Tier1 configuration with migration support
+        let tier1 = if cfg.tier1.enabled {
+            // Use the new Tier1 config
+            Some(Tier1 {
+                connect_interval: cfg.tier1.connect_interval,
+                new_connections_per_attempt: cfg.tier1.new_connections_per_attempt,
+                advertise_proxies_interval: time::Duration::minutes(15),
+                enable_inbound: cfg.tier1.enable_inbound,
+                enable_outbound: cfg.tier1.enable_outbound,
+            })
+        } else {
+            // For backwards compatibility, check if old configuration has tier1 enabled
+            // This is for migration support
+            None
```

Comment:
The implementation doesn't seem to match the comment above. If we can provide a smooth migration experience that would certainly be great, especially if we can log a warning advising that the experimental config is being deprecated.

However, in practice I don't think we expect or recommend any validator to use a non-standard config for Tier1 network, so I am OK with a more aggressive change.